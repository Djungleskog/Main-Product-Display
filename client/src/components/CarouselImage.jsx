import React from 'react';

let CarouselImage = function(props) {

  let currentImage = props.currentImageIndex === props.index;
  let difference = props.index - props.currentImageIndex;

  var clickHandler = function(event) {
    props.zoomClickHandler();
    console.log('clickevent: ', event);
  };

  var handleMouseEnter = function(event) {
    if (props.isZoomed) {
      console.log('mouseEnter event: ', event);
    }
  };

  var handleMouseMove = function(event) {
    if (props.isZoomed) {
      console.log('mouseMove event: ', event.pageX, event.pageY);
    }
  };

  return (
    <div onClick={clickHandler} className={currentImage ? 'carouselImageContainer' : 'notCurrentImageContainer'} style={currentImage ? {} : {position: 'absolute', left: difference < 0 ? difference * 2000 : '', right: difference > 0 ? difference * 2000 : ''} }>
      <img onMouseEnter={handleMouseEnter} onMouseMove={handleMouseMove} className={props.isZoomed ? 'currentImage zoomedImage' : 'currentImage'} src={props.currentImage}></img>
    </div>
  );
};

export default CarouselImage;