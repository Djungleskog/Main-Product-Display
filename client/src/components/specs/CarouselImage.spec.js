import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

import CarouselImage from '../CarouselImage.jsx';

describe('CarouselImage', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<CarouselImage />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});