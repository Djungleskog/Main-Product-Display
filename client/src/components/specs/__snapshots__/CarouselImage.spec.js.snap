// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CarouselImage snapshot renders 1`] = `
<div
  className="carouselImageContainer"
  onClick={[Function]}
  style={Object {}}
>
  <img
    className="currentImage"
    onMouseMove={[Function]}
    style={Object {}}
  />
</div>
`;
